drivers/HwWatchdog.cpp

```diff
@@ -0,0 +1,89 @@
+
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef DEVICE_WATCHDOG
+
+#include "HwWatchdog.h"
+
+namespace mbed {
+rtos::Mutex HwWatchdog::_Mutex;
+mbed_error_status_t HwWatchdog::start()
+{
+
+     watchdog_status_t sts;
+    if (HW_WATCHDOG_TIMEOUT == 0) {
+        return MBED_ERROR_INVALID_ARGUMENT;
+    }
+
+    if (HW_WATCHDOG_TIMEOUT > max_timeout()) {
+        return MBED_ERROR_INVALID_ARGUMENT;
+    }
+
+    _Mutex.lock();
+    watchdog_config_t config;
+    config.timeout_ms = HW_WATCHDOG_TIMEOUT;
+
+    sts = hal_watchdog_init(&config);
+    if(sts != WATCHDOG_STATUS_OK ) {
+        _Mutex.unlock();
+        return MBED_ERROR_INITIALIZATION_FAILED;
+     }
+    else{
```

Comment:
You don't need an to put an `else` after a `return`

Reply:
I have removed the else.Thanks for review comment